### PR TITLE
[libc][math]fadd implementation

### PR DESCRIPTION
**Pull request by aaryanshukla**
- **[libc] math fadd**
- **[libc][math] implemented fadd**

### CODE REVIEW

File: libc/src/math/generic/CMakeLists.txt [hunk 1]
```diff
@@ -452,6 +452,19 @@ add_entrypoint_object(
...
 )
 
+add_entrypoint_object(
+  fadd
+  SRCS
+    fadd.cpp
+  HDRS
+    ../fadd.h
+  DEPENDS
+    libc.src.__support.macros.properties.types
```
Comment (lntue) [inline libc/src/math/generic/CMakeLists.txt:462]: `libc.src.__support.macros.properties.types` is not needed.

File: libc/src/math/generic/fadd.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#include "src/__support/macros/config.h"
+#include "src/__support/FPUtil/generic/add_sub.h"
+#include "src/__support/common.h"
+
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(float, fadd, (double x, double y)) {
+    return fputil::generic::add<float>(x,y);
+
```
Comment (lntue) [inline libc/src/math/generic/fadd.cpp:19]: nit: remove extra empty line

File: libc/config/linux/x86_64/entrypoints.txt [hunk 3]
```diff
@@ -396,6 +396,7 @@ set(TARGET_LIBM_ENTRYPOINTS
     libc.src.math.fabs
     libc.src.math.fabsf
     libc.src.math.fabsl
+    libc.src.math.fadd
```
Comment (lntue) [inline libc/config/linux/x86_64/entrypoints.txt:399]: Can you also add to other architectures under `linux`?

**Review by lntue — APPROVED** [review]

**Comment by lntue** [comment]
Can you also update the math status page?

**Review by lntue — APPROVED** [review]

File: libc/src/math/fadd.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/__support/macros/config.h"
+
+#ifndef LLVM_LIBC_SRC_MATH_FADD_H
+#define LLVM_LIBC_SRC_MATH_FADD_H
```
Comment (overmighty) [inline libc/src/math/fadd.h:12]: Header guards should precede everything except the file header with license information.

File: libc/src/math/generic/CMakeLists.txt [hunk 5]
```diff
@@ -452,6 +452,18 @@ add_entrypoint_object(
...
 )
 
+add_entrypoint_object(
+  fadd
+  SRCS
+    fadd.cpp
+  HDRS
+    ../fadd.h
+  DEPENDS
+    libc.src.__support.FPUtil.basic_operations
```
Comment (overmighty) [inline libc/src/math/generic/CMakeLists.txt:462]: Should depend on `libc.src.__support.FPUtil.generic.add_sub` instead.

**Pull request merged**